### PR TITLE
Fix get_non_fingerprinted with dashed filenames

### PR DESCRIPTION
**Pull request by cgriego**
`upload_files` uses `get_non_fingerprinted` to determine if any non-fingerprinted files need to be uploaded because there's a new fingerprinted file. It does this by determining that if `assets/image-digest.png` needs to be uploaded then `assets/image.png` should also be uploaded, if it exists. But it faultily would determine that if `assets/great-image-digest.png` needed to be uploaded then `assets/great.png` should be uploaded if it exists instead of `assets/great-image.png`.

### CODE REVIEW

**Comment by PikachuEXE** [comment]
Will try to review this week
On vacation last week

File: lib/asset_sync/storage.rb [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
 module AssetSync
   class Storage
-    REGEXP_FINGERPRINTED_FILES = /^(.*)\/([^-]+)-[^\.]+\.([^\.]+)$/
+    REGEXP_FINGERPRINTED_FILES = /\A(.*)\/(.+)-[^\.]+\.([^\.]+)\z/
```
Comment (PikachuEXE) [inline lib/asset_sync/storage.rb:7]: Any reason you changed the anchors as well?

[Ref for anchors](https://ruby-doc.org/core-3.0.0/Regexp.html#class-Regexp-label-Anchors)

Comment (cgriego) [reply lib/asset_sync/storage.rb:7]: The original anchors only anchored per line and not the whole string. While rare, it is possible to have file names with new lines.

Comment (PikachuEXE) [reply lib/asset_sync/storage.rb:7]: Is it possible to add a test case for file name with new line then?

Comment (cgriego) [reply lib/asset_sync/storage.rb]: Added. I originally left it out since it's such an extreme edge case, though it did flush out another issue.

**Comment by PikachuEXE** [comment]
Released in `2.13.1`
